src/util.ts

```diff
@@ -62,6 +62,27 @@ export function fractionBN(
   return targetBN.mul(numBN).div(denomBN);
 }
 
+const BN_1000 = new BN(1000, 10);
+
+/**
+ * Used to convert a base-10 number from GWEI to WEI. Can handle numbers with decimal parts
+ *
+ * @param n - The base 10 number to convert to WEI
+ * @returns - The number in WEI, as a BN
+ */
+export function gweiDecToWEIBN(n: number | string) {
+  const wholePart = Math.floor(Number(n));
+  const decimalPartMatch = Number(n)
+    .toFixed(3)
```

Comment:
Huh. So digits after the third are omitted :thinking: 

This seems fine, given that WEI isn't really divisible. But this is probably worth documenting at least, so it doesn't come as a surprise (e.g. "rounds down to the nearest WEI".).

Reply:
yes, except I realize now that a GWEI is 1,000,000,000 WEI (10^9), and not just 1000 WEI (10^3) as this logic would imply.
:facepalm: 

changing to `.toFixed(9)` should fix that.